src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CollectInsertSizeMetricsSpark.java

```diff
@@ -210,84 +209,73 @@ public boolean test(final GATKRead read){
     private static final class SVCustomReadFilter implements ReadFilter{
         private static final long serialVersionUID = 1L;
 
-        private final ReadFilter collapsedFilter;
+        private final ReadFilter combinedReadFilter;
 
-        public SVCustomReadFilter(final UseEnd whichEnd,
-                                  final boolean filterNonProperlyPairdReads,
-                                  final boolean filterUnmappedReads,
-                                  final boolean filterMateUnmappedReads,
-                                  final boolean filterDupReads,
+        public SVCustomReadFilter(final EndToUse whichEnd,
+                                  final boolean filterNonProperlyPairedReads,
+                                  final boolean filterDuplicatedReads,
                                   final boolean filterSecondaryAlignments,
                                   final boolean filterSupplementaryAlignments,
-                                  final int MQThreshold){
-
-                final UseEnd endVal = whichEnd;
+                                  final int     MQThreshold,
+                                  final int     mMQThreshold){
 
-                ReadFilter tempFilter = read -> 0!=read.getFragmentLength();
```

Comment:
Here I think you might want to include a READ_MAPPED filter, no? Most of these other checks you are doing don't make sense unless the read is mapped, so I think you might want to reassert that filter at the beginning of this one.